### PR TITLE
Added code for Securonix Release 200

### PR DESCRIPTION
**Pull request by scnx-knathani**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
- Added support for Incident Mirroring. With version, 2.0.0 users can mirror the incidents comments, attachments & states across the platforms(Securonix & XSOAR).

- Added support for fetching threats from Securonix which will fetch Securonix Threats (with related violations) as XSOAR Incidents.

- Added a configurable retry mechanism for fetching incidents and performing actions using commands.

- Improved implementation of the token re-generation logic, now will generate a token every 24 hours instead of generating it for each API call.

- Added support for CRUD operation on Securonix Whitelist.
**securonix-whitelists-get**
**securonix-whitelist-entry-list**
**securonix-whitelist-entry-add**
**securonix-whitelist-entry-delete**
**securonix-whitelist-create**

- Added support for CRUD operation on Securonix Lookup tables.
**securonix-lookup-table-config-and-data-delete**
**securonix-lookup-tables-list**
**securonix-lookup-table-entry-add**
**securonix-lookup-table-entries-list**
**securonix-lookup-table-create**

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [x] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/Securonix/Integrations/Securonix/test_data/result_constants.py [hunk 1]
```diff
@@ -207,3 +210,294 @@
...
+
+EXPECTED_GET_INCIDENT_ATTACHMENT_HISTORY_6_4 = [
+    {'Contents': '', 'ContentsFormat': 'text', 'Type': 3, 'File': 'test.txt',
+     'FileID': '6c8964a3-aebd-4a4e-9d12-63b08c82372f'}]
+
+EXPECTED_LIST_VIOLATION_DATA_6_4 = [
+        {
+            "Timeline_By_Month": "1672552800000",
+            "Resourcegroupname": "SNX-IEE-AEE-51",
+            "Eventid": "3ae64498-1b89-44d7-aa14-6dc72af814bf",
```
Comment (yucohen) [inline Packs/Securonix/Integrations/Securonix/test_data/result_constants.py:296]: are those real ID's? If so, lets change those to dummy values.

**Comment by efelmandar** [comment]
@scnx-knathani Thank you for submitting your contribution! I've reviewed your PR and have identified a few areas that I would like you to consider before merging. 

1. To avoid creating duplicate fields, please use existing incident fields from the "commonTypes" pack where possible. for example: `Tenant Name` field already exist in `CommonTypes` and can be used instead of creating new one
2. The current configuration of the "incidentTypes" is not using the auto-extract function, which is usually not recommended. it might be useful if you are sure the fields in the response won't contain indicators to enrich, but usually it only apply for specific fields
3. Since there are two incident types, we recommend adding a classifier to automate the incident type classification. This will help ensure accuracy and efficiency. 

We appreciate your efforts and look forward to your response.

**Review by yucohen — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please see my comments

File: Packs/Securonix/Integrations/Securonix/test_data/result_constants.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (scnx-knathani) [reply Packs/Securonix/Integrations/Securonix/test_data/result_constants.py]: Done, changed to dummy value.

File: Packs/Securonix/README.md [hunk 2]
```diff
@@ -1,2 +1,2 @@
-Note: Support for this pack was moved to Securonix as of 05/25/2022
-
+Note: Support for this pack was moved to Securonix as of 05/25/2022
+
```
[the remark this answers is not in the document]
Comment (scnx-knathani) [reply Packs/Securonix/README.md:11]: Updated description.

File: Packs/Securonix/Integrations/Securonix/Securonix.py [hunk 3]
```diff
@@ -1227,13 +3080,30 @@ def main():
     password = params.get('password')
     verify = not params.get('unsecure', False)
     proxy = demisto.params().get('proxy') is True
+    # Updating TOTAL_RETRY_COUNT to get user provided value
+    global TOTAL_RETRY_COUNT
+    TOTAL_RETRY_COUNT = arg_to_number(params.get('securonix_retry_count', '0'),  # type: ignore
```
[the remark this answers is not in the document]
Comment (scnx-knathani) [reply Packs/Securonix/Integrations/Securonix/Securonix.py:3086]: Agree, added validation for the same.

File: Packs/Securonix/Integrations/Securonix/Securonix.py [hunk 4]
```diff
@@ -177,10 +658,25 @@ def http_request(self, method, url_suffix, headers=None, params=None, response_t
...
                       f'Error Number: [{exception.errno}]\n' \
                       f'Message: {exception.strerror}\n' \
-                      f'Verify that the tenant parameter is correct' \
+                      f'Verify that the tenant parameter is correct ' \
                       f'and that you have access to the server from your host.'
             raise Exception(f'{err_msg}\n{exception}')
 
+        except requests.exceptions.RetryError as exception:
+            try:
+                reason = 'Reason: {}'.format(exception.args[0].reason.args[0])  # pylint: disable=no-member
```
[the remark this answers is not in the document]
Comment (scnx-knathani) [reply Packs/Securonix/Integrations/Securonix/Securonix.py]: Done, Changed to f-strings wherever required.

**Comment by scnx-knathani** [comment]
> @scnx-knathani Thank you for submitting your contribution! I've reviewed your PR and have identified a few areas that I would like you to consider before merging.
> 
> 1. To avoid creating duplicate fields, please use existing incident fields from the "commonTypes" pack where possible. for example: `Tenant Name` field already exist in `CommonTypes` and can be used instead of creating new one
> 2. The current configuration of the "incidentTypes" is not using the auto-extract function, which is usually not recommended. it might be useful if you are sure the fields in the response won't contain indicators to enrich, but usually it only apply for specific fields
> 3. Since there are two incident types, we recommend adding a classifier to automate the incident type classification. This will help ensure accuracy and efficiency.
> 
> We appreciate your efforts and look forward to your response.

Hey @efelmandar,

1. That's a good catch, used the existing Tenant Name field available in Common Types packs.
2. Updated incident types to auto-extract all the available indicators if any.
3. In this case, Both incident types are fetched from different APIs, having a different set of fields and not a single common field that's why we have not used a classifier for bifurcating incident types.

**Comment by scnx-knathani** [comment]
Hello @efelmandar @yucohen, 

We have resolved/replied to all the review comments. Please let us know if there are more changes required.

**Comment by scnx-knathani** [comment]
Hello @efelmandar @yucohen @edik24 

We have resolved all the review comments based on your comments in yesterday's demo. 
Please let us know if there are any more suggestions.

**Review by yucohen — APPROVED** [review]

**Pull request merged**